dotnet/src/SemanticKernel.Core/SemanticKernel.Core.csproj

```
@@ -7,7 +7,7 @@
     <TargetFrameworks>net8.0;netstandard2.0</TargetFrameworks>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
-    <NoWarn>$(NoWarn);SKEXP0001</NoWarn>
+    <NoWarn>$(NoWarn);SKEXP0001,SKEXP0120</NoWarn>
```

I suggest removing this and making this visible in the source code, so we know if any of our non-experimental public APIs are currently exposed to experimental stuff.

```suggestion
    <NoWarn>$(NoWarn);SKEXP0001</NoWarn>
```

---

What would be the value in doing that? Should we also drop SKEXP0001 from the exception list and wrap all usage sites of the experimental stuff with #pragma in more than 100 places?

---

### Interesting topic for discussion outside of this PR

This will increase our awareness on the Experimental dependency, , some extra information to be considered for this to be practical also need to add `.editorconfig` to be a `Error/Warning` on [IDE0079 - Remove unnecessary suppression](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0079)